### PR TITLE
feat(plugin-sheet): Text wrap style

### LINKED ISSUE #8101: [spreadsheet] Soft wrapping formatting

### PR DESCRIPTION
**Pull request by thure**
This pr:
- resolves #8101
- fixes a variety of issues around Sheet’s Toolbar.

### CODE REVIEW

File: packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx [hunk 1]
```diff
@@ -214,7 +219,7 @@ const Alignment = () => {
             ({ range, key }) => key === alignKey && inRange(rangeFromIndex(model.sheet, range), cursor),
           )?.value
         : undefined,
-    [cursor, model.sheet.ranges],
```
Comment (wittjosiah) [inline packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx:217]: this doesn't need to be recreated if the ranges change?

File: packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx [hunk 2]
```diff
@@ -235,12 +240,17 @@ const Alignment = () => {
...
+const styleOptions: ButtonProps<StyleValue>[] = [
+  { value: 'highlight', icon: 'ph--highlighter--regular' },
+  { value: 'softwrap', icon: 'ph--paragraph--regular' },
+];
 
 const Styles = () => {
   const { cursor, model } = useSheetContext();
   const { onAction } = useToolbarContext('Styles');
   const { t } = useTranslation(SHEET_PLUGIN);
+  // TODO(thure): Why is neither model.sheet nor model.sheet.ranges not usable as a memo dependency?
```
Comment (wittjosiah) [inline packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx:252]: What happens without this?

File: packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx (hunk 1 above)
Comment (thure) [reply packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx:217]: I think I was trying to figure out reactivity here, will revert and create a new issue to address toolbar reactivity.

File: packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx (hunk 2 above)
Comment (thure) [reply packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx:252]: The toolbar item isn’t reactive even though the range is added.

Comment (thure) [reply packages/plugins/plugin-sheet/src/components/Toolbar/Toolbar.tsx:252]: May just remove the `useMemo`…

**Pull request merged**